cmd/vm-informant/main.go

```diff
@@ -64,3 +95,55 @@ func main() {
 		klog.Fatalf("Server failed: %s", err)
 	}
 }
+
+// runRestartOnFailure repeatedly calls this binary with the same flags, but with 'auto-restart'
+// removed.
+//
+// We execute ourselves as a subprocess so that it's possible to appropriately cleanup after
+// termination by various signals (or an unhandled panic!). This is worthwhile because we *really*
+// don't want to leave the cgroup frozen while waiting to restart.
+func runRestartOnFailure(args []string, cleanupHooks []func()) {
+	selfPath := os.Args[0]
+
+	minWaitDuration := time.Second * 5
+
+	for {
+		minWait := time.After(minWaitDuration)
```

Comment:
Don't think that gets us anything here

Reply:
I mean, ideally you'd have some kind of lifecycle (so that it would handle shut down/restart requests, upgrades, etc.) management, in which case we could handle signals more quickly? (which would involve having contexts, and not using this like a sleep, but it can all be changed later?

Reply:
Yeah, fair. Wrote half a comment re: systemd, forgot to finish it but now it's there - seems kind of related. Mostly this PR is just replicating 5 lines of bash, really just so we can keep the cgroups clean. Better to keep it simple for now I think.